### ISSUE TITLE
`prefer-spread` false positive for tRPCs `procedure.concat`

### ISSUE THREAD

**Issue by DesselBane** [sindresorhus/eslint-plugin-unicorn#2641]
In tRPC one can concatinate multiple middlewares ([docs](https://trpc.io/docs/server/middlewares#concat))

This usually looks like this:

```typescript
const foo = t.procedure.concat(myMiddleware)

// Or
const bar = myCustomProcedure.concat(myMiddleware)
```

Both variants get flagged by the `unicorn/prefer-spread` rule even though they are not arrays and the rule makes no sense in this case.
